### PR TITLE
The response body is compressed when the content type of the response in types.

### PR DESCRIPTION
**Pull request by BerryWang1996**
Motivation:

The response body is compressed when the content type of the response in types.

Modifications:

Add parameter named "types" in constructor, and add related code in method named "beginEncode".

Result:

If content type in response body is not contained by parameter named types, the response body will not be compressed.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@BerryWang1996 what is the reason for the PR ? It seems like it not changes anything in terms of functionality.

**Comment by normanmaurer** [comment]
@BerryWang1996 why you need this?

**Comment by BerryWang1996** [comment]
@normanmaurer Download file failed when use HttpContentCompressor. The binary stream should not be compressed...I think set content type is flexible than before...

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java [hunk 1]
```diff
@@ -103,6 +108,38 @@ public HttpContentCompressor(int compressionLevel, int windowBits, int memLevel)
...
+     * @param memLevel
+     *        How much memory should be allocated for the internal compression
+     *        state.  {@code 1} uses minimum memory and {@code 9} uses maximum
+     *        memory.  Larger values result in better and faster compression
+     *        at the expense of memory usage.  The default value is {@code 8}
+     * @param contentSizeThreshold
+     *        The response body is compressed when the size of the response
+     *        body exceeds the threshold. The value should be a non negative
+     *        number. {@code 0} will enable compression for all responses.
+     * @param types
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java]: We should make this `String... types`

**Comment by normanmaurer** [comment]
@BerryWang1996 please add a unit test

**Comment by normanmaurer** [comment]
Closing due inactivity

**Pull request closed without merging**